crates/bevy_render/src/mesh/allocator.rs

```diff
@@ -588,6 +588,8 @@ impl MeshAllocator {
         }
 
         for empty_slab in empty_slabs {
+            self.slab_layouts
+                .retain(|_, slab_ids| !slab_ids.contains(&empty_slab));
```

Comment:
Actually looking again I don't think the current fix is right, I misread the `.retain` to be on the inner vecs instead of the hashmap when I looked yesterday. There could be other slabs with the same layout, and those should remain.

This makes performance a bit more iffy, if it ever becomes a problem we could use `HashSet` instead of the vecs.

```suggestion
            self.slab_layouts.iter_mut().for_each(|(_, slabs)| {
                slabs.retain(|&slab_id| slab_id != empty_slab);
            });
```

Reply:
Can we do
```rust
let idx = slabs.iter().position(|&slab_id| slab_id == empty_slab);
                if let Some(idx) = idx {
                    slabs.remove(idx);
                }
                ```
to short circuit in the event we find the slab? Are they guarunteed to be unique within the vec?

Reply:
Yeah they're guaranteed to be unique and in the range 0..next_slab_id, so that would work.

Reply:
Thanks. This is what I meant to write :sweat_smile: